src/Serializer/Normalizer/ArrayNormalizer.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcing\Serializer\Normalizer;
+
+use function array_map;
+use function is_array;
+
+class ArrayNormalizer implements Normalizer
+{
+    public function __construct(private readonly Normalizer $normalizer)
+    {
+    }
+
+    public function normalize(mixed $value): ?array
+    {
+        if ($value === null) {
```

Comment:
Im not sure about the nullability part. Maybe this should be configured so that the normalizer throw if nullability is not allowed? I dont know, if this is somehow better for the user instead of maybe a hard crash, if the property does not allow `null`. Do we have a testcase for this maybe?

Reply:
I don't think we should ensure that the value is valid. Whether it's nullable or not, or if you can put a DateTimeNormalizer on a boolean property. We don't do validation here. 

In Doctrine you can also set a column type that doesn't match your property type:

https://github.com/doctrine/dbal/blob/3.3.x/src/Types/DateTimeImmutableType.php

But I will improve the error handling 👍